src/inference/src/threading/ie_istreams_executor.cpp

```diff
@@ -46,51 +52,65 @@ int IStreamsExecutor::Config::GetDefaultNumStreams() {
-            config._big_core_streams = num_big_cores / config._threads_per_stream_big;
+            big_core_streams = num_big_cores / threads_per_stream_big;
+            threads_per_stream_small = 0;
+        } else if (num_small_cores < threads_per_stream_small) {
+            small_core_streams = 1;
+            threads_per_stream_small = num_small_cores;
+            threads_per_stream_big = threads_per_stream_small;
+            big_core_streams =
+                logic_core ? num_big_cores / threads_per_stream_big * 2 : num_big_cores / threads_per_stream_big;
         } else {
-            config._small_core_streams = num_small_cores / config._threads_per_stream_small;
+            small_core_streams = num_small_cores / threads_per_stream_small;
         }
     } else if (stream_mode == AGGRESSIVE) {
-        config._big_core_streams = num_big_cores;
-        config._small_core_streams = num_small_cores / 2;
-        config._threads_per_stream_big = num_big_cores / config._big_core_streams;
-        config._threads_per_stream_small = num_small_cores == 0 ? 0 : num_small_cores / config._small_core_streams;
+        big_core_streams = logic_core ? num_big_cores_total : num_big_cores;
+        small_core_streams = logic_core ? num_small_cores : num_small_cores / 2;
+        threads_per_stream_big = logic_core ? num_big_cores_total / big_core_streams : num_big_cores / big_core_streams;
+        threads_per_stream_small = num_small_cores == 0 ? 0 : num_small_cores / small_core_streams;
     } else if (stream_mode == LESSAGGRESSIVE) {
-        config._big_core_streams = num_big_cores / 2;
-        config._small_core_streams = num_small_cores / 4;
-        config._threads_per_stream_big = num_big_cores / config._big_core_streams;
-        config._threads_per_stream_small = num_small_cores == 0 ? 0 : num_small_cores / config._small_core_streams;
+        big_core_streams = logic_core ? num_big_cores_total / 2 : num_big_cores / 2;
+        small_core_streams = logic_core ? num_small_cores / 2 : num_small_cores / 4;
+        threads_per_stream_big = logic_core ? num_big_cores_total / big_core_streams : num_big_cores / big_core_streams;
+        threads_per_stream_small = num_small_cores == 0 ? 0 : num_small_cores / small_core_streams;
     } else {
         IE_THROW() << "Wrong stream mode to get num of streams: " << stream_mode;
     }
-    config._small_core_offset = num_small_cores == 0 ? 0 : num_big_cores * 2;
-    return config._big_core_streams + config._small_core_streams;
+
+    config[CONFIG_KEY(BIG_CORE_STREAMS)] = std::to_string(big_core_streams);
+    config[CONFIG_KEY(SMALL_CORE_STREAMS)] = std::to_string(small_core_streams);
+    config[CONFIG_KEY(THREADS_PER_STREAM_BIG)] = std::to_string(threads_per_stream_big);
+    config[CONFIG_KEY(THREADS_PER_STREAM_SMALL)] = std::to_string(threads_per_stream_small);
+    config[CONFIG_KEY(SMALL_CORE_OFFSET)] = std::to_string(num_small_cores == 0 ? 0 : num_big_cores * 2);
```

Comment:
Do I understand correctly that with "num_big_cores * 2" we assume BIG cores always have hyper-threads? Is it always true now and will be true for future generations?

Reply:
Hyper-threading can be disabled in BIOS by users in certain use-cases. it might be better to determine at runtime whether Hyper-Threading is available, enabled or disabled.